### PR TITLE
[Middleware] Added API calls + refactored rendering for "today" todo list

### LINKED ISSUE #32: Make a new todo for each day based on time

### PR DESCRIPTION
**Pull request by stevnzhang**
Added today logic for getting the current date's todos. If today's todo list exists, then use that one, otherwise, make a new one and put it in the database.

Refactored some of the pages to take in a today obj for updating the todo list for current date.

Resolves #32

### CODE REVIEW

File: EXP-Client/todo/TodoList.js [hunk 1]
```diff
@@ -26,15 +26,17 @@ export default function TodoList({ todos, setIsPanelActive, setViewingTodo }) {
...
           </Pressable>
           {todos.map(({ name, deadline, priority, completed }, index) => (
-            <DataTable.Row key={name} style={styles.entry}>
+            <Pressable key={name} onPress={() => setViewingTodo({ name, deadline, priority, completed })}>
+              <DataTable.Row style={styles.entry}>
               <DataTable.Cell style={{ justifyContent: 'center', flex: 1 }}>
                 <Checkbox value={completed} />
               </DataTable.Cell>
-              <Pressable style={{ flex: 5 }} onPress={() => setViewingTodo({ name, deadline, priority, completed })}>
+              <DataTable.Cell style = {{ flex: 5 }}>
```
Comment (emmayjiang) [inline EXP-Client/todo/TodoList.js:34]: steben you broke the alignment :(

image

**Comment by emmayjiang** [comment]
Wasn't sure how to test the logic but appearances LGTM besides the nitpick I commented earlier.

Side note unrelated to the contents of this PR, but we should probably also update this date 
image

**Review by emmayjiang — APPROVED** [review]

**Comment by stevnzhang** [comment]
> Wasn't sure how to test the logic but appearances LGTM besides the nitpick I commented earlier.
> 
> Side note unrelated to the contents of this PR, but we should probably also update this date image

Yup, I'll update this in my next PR

File: EXP-Client/todo/TodoList.js (hunk 1 above)
Comment (stevnzhang) [reply EXP-Client/todo/TodoList.js:34]: Oh no, I'll see if I can fix this

**Review by emmayjiang — APPROVED** [review]
LGTM

**Pull request merged**